### PR TITLE
zerotier: update to version 1.2.4

### PR DESCRIPTION
**Pull request by mwarning**
Maintainer: me
Compile tested: LEDE 17.01.1
Run tested: ar71xx

Description:
Zerotier now depends on libstd, libminiupnpc and libnatpmp.
uClibc++ does not work anymore.

### CODE REVIEW

**Comment by mwarning** [comment]
ZT does successfully connect to the open EARTH network.

Selftest result:
```
root@LEDE:~# zerotier-selftest 
[info] sizeof(void *) == 4
[info] sizeof(NetworkConfig) == 431368
[other] Testing C++ exceptions... OK
[other] Testing/fuzzing Dictionary... PASS (junk value to prevent optimization-out of test: -16878)
[crypto] getSecureRandom: df83557cea91fac47159043dc940cc36f19803974a85e62b5885690b02a31ee072c349ebaf8cc71bed226591f67d4c1866d0be3b9c22e684f0eeeb4864477d91
[crypto] getSecureRandom: e17b490d05529aa0148ac9ee447636f30356a721a6f5ae7d209e21bccb34d4efc63c0315bc2e70d404ac2f34b33cbedce6a3789ba02d057238806a5018a47bfa
[crypto] getSecureRandom: a8e6ccb24bf83e0d783a88c0da441a53375dcbe78cd1f67f309c6b81bb005d750701cea4e9a590ac29c04f8091cc19bd65f2a4fc439b39c81a168c75c96c1dd7
[crypto] Testing Salsa20... PASS
[crypto] Salsa20 SSE: DISABLED
[crypto] Benchmarking Salsa20/12... 10.6603 MiB/second (f31ddbcdb3b69d0a5e19ba94a7e2facc)
[crypto] Benchmarking Salsa20/20... 7.39742 MiB/second (24c619994ae7f1549136169c9072f3d7)
[crypto] Testing SHA-512... PASS
[crypto] Testing Poly1305... PASS
[crypto] Benchmarking Poly1305... 12.8914 MiB/second
[crypto] Testing C25519 and Ed25519 against test vectors... PASS
[crypto] Testing C25519 ECC key agreement... PASS
[crypto] Benchmarking C25519 ECC key agreement... 97.36ms per agreement.
[crypto] Testing Ed25519 ECC signatures... PASS
[packet] Testing Packet encoder/decoder... (compressed: 77, decompressed: 1116) PASS
[identity] Validate known-good identity... PASS (1538.9ms per validation)
[identity] Validate known-bad identity... PASS (i.e. it failed)
[identity] Generate identity... (took 17024ms): 58d94fb780:0:b2ebffb84ce54e2341672f14afd3157f5032fe1bc99304452d4416cdeaeb7c1ab4e3a1e6adcb616c24e55b253b560455ea2e15252786b0c07d9565033aaccc14:2783b6c4f1032cb39bdec5e2e7ce437b89a25c1f7b3b39a2e4ab87bb3f6dded3b94482af7580c4c0840d28abba008eb3831ebdf44f7751b698d574dfe6f3845e
[identity] Locally validate identity: PASS
[identity] Generate identity... (took 14390ms): 57bd5e3cb2:0:765dca038313caf20857bdb9ff65d0b8d7b8dfe8fe463f7388468763ce7d411de7ae8cc9889ce4515efc5e4709ef216599b7f009c499d92d7f0d155c3b81c926:a32c3993d42c3030d0cd7a46ee7adcf59c2fc9959f8e4fa5f73988338f07f6b5406742653f998e13a02cf9591c51bbbefc73645363193d58dacc783c24b116f0
[identity] Locally validate identity: PASS
[identity] Generate identity... (took 12203ms): c5e19d67e3:0:144bbed5fc6f27ba438499e69037ba3d9a12cc00f53e4a85f950197921a45a3fe9d1b6e306b604b72b0ef69e0c1980fb5ec80a3b042a1e4e71f91eaf9ef0b93b:a23507417cede757ddbd54b03bd5ed14ade95493f8c627642166666b1b3f431f41c8ec61c1e6af0430f7c4714d44dbce58fa8668666a859a7db95ed2687e919f
[identity] Locally validate identity: PASS
[identity] Generate identity... (took 32046ms): 4cdfa326bb:0:05f2e29db15d3c17428fb9abba6467c7a74a25fae2a0e12b390f8b470108a3706f7cdfedd0d89bd578ad38167085882dab34eb78c5ddb3f7ea2f7d660b9af82c:32bb5bfdf47ab4bd612b630c0d7bb9d50eff1da95c9b782dad16d3c4239215489ff7951b467984be2e083061945f25fad843c7c15e407580ecb9df9909601369
[identity] Locally validate identity: PASS
[identity] Serialize and deserialize (w/private): PASS
[identity] Serialize and deserialize (no private): PASS
[identity] Serialize and deserialize (ASCII w/private): PASS
[identity] Serialize and deserialize (ASCII no private): PASS
[certificate] Generating identity to act as authority... 1236bcd670
[certificate] Generating identities A and B... a53f65aed8, 0e4d2c0c0c
[certificate] Generating certificates A and B...
[certificate] Signing certificates A and B with authority...
[certificate] A agrees with B and B with A... yes, yes.
[certificate] Generating two certificates that should not agree...
[certificate] A agrees with B and B with A... no, no.
[phy] Creating phy endpoint...
[phy] Binding UDP listen socket to 127.0.0.1/60002... OK
[phy] Binding TCP listen socket to 127.0.0.1/60002... OK
[phy] Testing UDP send/receive... got 10000 packets, OK
[phy] Testing TCP... got 10 connect successes, 2 failures, and 10000000 bytes, OK
```

Open Questions:
- does PKG_MIRROR_MD5SUM need to be updated despite PKG_SOURCE_VERSION?

**Comment by hnyman** [comment]
> does PKG_MIRROR_MD5SUM need to be updated despite PKG_SOURCE_VERSION?

PKG_MIRROR_HASH enables the build system to reuse already fetched archive copies at the local source mirror instead of always fetching new sources with git.

So, please refresh the PKG_MIRROR_HASH value (note the change from PKG_MIRROR_MD5SUM to PKG_MIRROR_HASH). Just calculate the sha256 from your local .tar.xz copy (that has been downloaded by the Makefile).

**Comment by mwarning** [comment]
@hnyman PKG_MIRROR_HASH has been updated

File: net/zerotier/Makefile [hunk 1]
```diff
@@ -63,7 +58,7 @@ define Package/zerotier/install
 	$(INSTALL_DIR) $(1)/usr/bin
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/zerotier-one $(1)/usr/bin/
 	$(LN) zerotier-one $(1)/usr/bin/zerotier-cli
-	$(LN) zerotier-one $(1)/usr/bin/zerotier-idtool
+	$(LN) zerotier-one $(1)/usr/bin//zerotier-idtool
```
Comment (hnyman) [inline net/zerotier/Makefile]: The double "//" in path looks like a typo.

**Comment by mwarning** [comment]
@hnyman fixed

**Pull request merged**